[PyOV] Add README.md and align structure-showing style (#14093)
openvinotoolkit/openvino: src/bindings/python/README.md

# OpenVINO Python API

> OpenVINO Python API is a part of OpenVINO library. The component is responsible for:

Reviewer: ```suggestion
OpenVINO Python API is a part of the OpenVINO library. The component is responsible for:
```

* Bindings of OpenVINO - allowing users to utilize OpenVINO library in their Python code. Python API provides bindings to basic and advanced APIs from OpenVINO Runtime.
* Extending OpenVINO with pythonic features - on top of direct translations from C++, Python API component:
    * Adds specific extensions to support numpy-based data.
    * Provides support for external frameworks inputs.
    * Provides shortcuts and helpers with more pythonic design.
    * Allows to apply advanced concepts like shared memory to take full advantage of OpenVINO.

> OpenVINO Python API uses [common codestyle checks](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/docs/contributing.md#contribution-guidelines-and-best-practices) that are adjusted to project needs.

Reviewer: ```suggestion
OpenVINO Python API uses [the common codestyle checks](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/docs/contributing.md#contribution-guidelines-and-best-practices) which are adjusted to project needs.
```

## Key contacts

> If you have any questions, feature requests or want us to review your PRs, send us a message or ping us on GitHub via [openvino-ie-python-api-maintainers](https://github.com/orgs/openvinotoolkit/teams/openvino-ie-python-api-maintainers). You can always directly contact everyone from this group.

Reviewer 2: nitpick: If I remember right you cannot write a message on github group :)

Author: Most of us have corporate emails shown in their profiles, thus I pointed to direct contact as an option.

## Components

OpenVINO PYTHON API has the following structure:

* [docs](./docs/) - folder that contains developer documentation and code examples.
* [src](./src/) - folder with all source files for Python API.
    * [src/compatibility](./src/compatibility/) - sources for compatibility API, including older modules like `ngraph` and `openvino.inference_engine`.

>     * [src/compatibility](./src/openvino/) - Python sources for current API.

Reviewer: the directories have the same name but the links are different, need to change the second one?

Author: Great catch!

    * [src/pyopenvino](./src/pyopenvino/) - C++ sources for current API.
* [tests](./tests/) - tests directory for current API.
* [tests_compatibility](./tests_compatibility/) - tests directory for compatibility API.

> * [thirdparty](./thirdparty/) - folder that contains thirdparty modules like `pybind11`.

Reviewer: ```suggestion
* [thirdparty](./thirdparty/) - folder that contains third-party modules like `pybind11`.
```

> * [wheel](./wheel/) - wheel specific directory, contains all specifc requirements and files used during wheel creation.

Reviewer: ```suggestion
* [wheel](./wheel/) - wheel-specific directory that contains all specific requirements and files used during wheel creation.
```

# OpenVINO Python API

OpenVINO Python API is a part of OpenVINO library. The component is responsible for:

* Bindings of OpenVINO - allowing users to utilize OpenVINO library in their Python code. Python API provides bindings to basic and advanced APIs from OpenVINO Runtime.
* Extending OpenVINO with pythonic features - on top of direct translations from C++, Python API component:
    * Adds specific extensions to support numpy-based data.
    * Provides support for external frameworks inputs.
    * Provides shortcuts and helpers with more pythonic design.

>     * Allows to apply advanced concepts like shared memory to take full advantage of OpenVINO.

Reviewer 2: Do we have any code styles and something like that for python?

Author: I will add that if necessary. However we are following adjusted style checks and everything is described in [contributing.md](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/docs/contributing.md), so link against it is enough?

Author: Linked with current state.

## Key contacts

> If you have any questions, feature requests or want us to review your PRs, send us a message or ping us on Github.

Reviewer 3: ```suggestion
If you have any questions, feature requests or want us to review your PRs, send us a message or ping us on GitHub.
```

Author: Got it!

Team members and main responsibilities:

* [Jan Iwaszkiewicz](https://github.com/jiwaszki) (e-mail: jan.iwaszkiewicz@intel.com)  
Techincal Lead: questions, issues and feature requests, memory and GIL related questions
* [Anastasia Kuporosova](https://github.com/akuporos) (e-mail: anastasia.kuporosova@intel.com)  
Developer: building of the component, general bindings
* [Przemyslaw Wysocki](https://github.com/p-wysocki) (e-mail: przemyslaw.wysocki@intel.com)  

> Developer: project requirements, general bindings

Reviewer 2: Please use github python maintainer instead of list of persons

## Components

```

> openvino/src/bindings/python

Reviewer 2: We already exists here :)
Maybe we can use a usual list as we have for other components?

Author: I see both options as equal for developers. However, if it is general practice across components, sure I will edit this.

Author: Done

├── docs                    <-- Developer documentation
│   └── examples            <-- Developer code examples
├── README.md               <-- This file:)
├── src
│   ├── compatibility       <-- Sources for compatibility API
│   ├── openvino            <-- Python sources for current API
│   └── pyopenvino          <-- C++ sources for current API
├── tests
├── tests_compatibility
├── thirdparty              <-- Thirdparty libraries
│   └── pybind11
└── wheel                   <-- Wheel specific directory
```

## Tutorials

> If you want to contribute to OpenVINO Python API, here is the list of learning materials and project's guidelines:

Reviewer 3: ```suggestion
If you want to contribute to OpenVINO Python API, here is the list of learning materials and project guidelines:
```

Author: Done

> * [How to contribute](docs/contributing.md)

Reviewer 2: Do we have a plan to change and review these files?
At the current moment it looks strange:
<img width="1158" alt="Screenshot 2022-11-18 at 16 24 46" src="https://user-images.githubusercontent.com/4737252/202704491-776ba088-0511-4a9c-a022-2308942e51d2.png">

* [How to extend OpenVINO Python API](docs/code_examples.md)
* [How to test OpenVINO Python API](docs/test_examples.md)
* [How to upgrade local Python version](docs/python_version_upgrade.md)

## See also

Related pages and articles:

> * [pybind11 repository](https://github.com/pybind/pybind11)

Reviewer 2: I propose to add links on high level documents as we have for other components.
